sessionize/app/src/main/java/co/touchlab/sessionize/NotificationsApiImpl.kt

```diff
@@ -1,75 +1,60 @@
 import android.content.Context
 import android.content.Intent
-import android.content.IntentFilter
 import android.os.Build
-import androidx.core.app.NotificationCompat
-import androidx.core.app.NotificationManagerCompat
 import app.sessionize.touchlab.lib.R
 import co.touchlab.sessionize.api.NotificationsApi
 import co.touchlab.sessionize.platform.AndroidAppContext
 import co.touchlab.sessionize.platform.NotificationsModel.setNotificationsEnabled
 
-
 class NotificationsApiImpl : NotificationsApi {
 
-    private val notificationPublisher: BroadcastReceiver = NotificationPublisher()
-
     override fun createLocalNotification(title:String, message:String, timeInMS:Long, notificationId: Int, notificationTag: String) {
-        // Building Notification
-        val channelId = AndroidAppContext.app.getString(R.string.notification_channel_id)
-        val builder = NotificationCompat.Builder(AndroidAppContext.app, channelId)
-                .setSmallIcon(R.drawable.baseline_insert_invitation_24)
-                .setContentTitle(title)
-                .setContentText(message)
-                .setWhen(timeInMS)
-                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
-                .setCategory(NotificationCompat.CATEGORY_REMINDER)
-
-
-        // Building Intent wrapper
-        val intent = Intent().also { intent ->
-            intent.action = AndroidAppContext.app.getString(R.string.notification_action)
-            intent.putExtra(NotificationPublisher.NOTIFICATION_ID, notificationId)
-            intent.putExtra(NotificationPublisher.NOTIFICATION_TAG, notificationTag)
-            intent.putExtra(NotificationPublisher.NOTIFICATION, builder.build())
-            val componentName = ComponentName(AndroidAppContext.app, NotificationPublisher::class.java)
-            intent.component = componentName
-        }
-
-        val pendingIntent = PendingIntent.getBroadcast(AndroidAppContext.app, notificationId, intent, PendingIntent.FLAG_CANCEL_CURRENT)
 
         print("Local $notificationTag Notification Created at $timeInMS: $title - $message \n")
 
-        // Scheduling Intent
-        val alarmManager = AndroidAppContext.app.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+        val pendingIntent = createPendingIntent(notificationId, notificationTag, NotificationPublisher.NOTIFICATION_ACTION_CREATE, title, message)
+        val alarmManager = AndroidAppContext.app.getSystemService(Activity.ALARM_SERVICE) as AlarmManager
         alarmManager.set(AlarmManager.RTC_WAKEUP, timeInMS, pendingIntent)
     }
 
-    override fun cancelLocalNotification(notificationId: Int, notificationTag: String) {
-        with(NotificationManagerCompat.from(AndroidAppContext.app)) {
-            this.cancel(notificationTag,notificationId)
-            print("Cancelling Local $notificationTag Notification")
+    override fun cancelLocalNotification(notificationId: Int, notificationTag: String, withDelay: Long?) {
```

Comment:
Rather than making withDelay nullable, you could make it optional with 0 as a default
`withDelay: Long = 0`